submissions/kovalov/frogger/app.js

```diff
@@ -0,0 +1,120 @@
+class Enemy {
+  constructor(x, y, speedX, fieldWidth, player) {
+    this.sprite = 'images/enemy-bug.png';
+    this.fieldWidth = fieldWidth;
+    this.x = x;
+    this.y = y;
+    this.speedX = speedX;
+    this.height = 40;
+    this.width = 83;
+    this.player = player;
+  }
+
+  checkCollision() {
+    if (
+      this.x + this.width > this.player.x &&
+      this.player.x + this.player.width > this.x &&
+      this.player.y > this.y - this.height &&
+      this.player.y - this.player.height < this.y
+    ) {
+      this.player.resetPosition();
+    }
+  }
+
+  update(dt) {
+    this.checkCollision();
+
+    if (this.x >= this.fieldWidth) {
+      this.x = 0;
+    }
+
+    this.x += dt * this.speedX;
+  }
+
+  render() {
+    ctx.drawImage(Resources.get(this.sprite), this.x, this.y);
+  }
+}
+
+class Player {
+  constructor(params) {
+    this.sprite = 'images/char-boy.png';
+    this.fieldWidth = params.fieldWidth;
+    this.fieldHeight = params.fieldHeight;
+    this.x = params.fieldWidth / 2;
+    this.y = params.fieldHeight;
+    this.speedX = params.speedX;
+    this.speedY = params.speedY;
+    this.height = 40;
+    this.width = 40;
+    this.score = 0;
+    this.scoreElement = document.querySelector('.score__number');
+  }
+
+  update(dt) {
+    if (this.y < 0) {
+      this.scoreElement.innerHTML = `${++this.score}`;
```

Comment:
Well designed class shouldn't interact with an external world directly, i.e. side effects aren't a good practice unless the class is designed specifically for this purpose and it is its specialty.